lib/SILOptimizer/Transforms/DeadObjectElimination.cpp

```diff
@@ -224,6 +269,28 @@ static bool canZapInstruction(SILInstruction *Inst, bool acceptRefCountInsts) {
   if (isa<DestroyAddrInst>(Inst))
     return true;
 
+  // As a special rule, a string interpolation is eliminatable if its result is
+  // unused, even though this theoretically could remove the side effects of
+  // StringInterpolationProtocol methods or the string-conversion witnesses they
+  // would have called if not eliminated. Mutating methods this rule applies to
+  // should be marked with @_semantic("interpolation.append"); they will only
+  // allow their self parameter to be eliminated.
+  if (auto *Apply = dyn_cast<ApplyInst>(Inst))
+    if (appendsToAllocatedInterpolation(Apply, AllocInst))
+      return true;
+
+  // We can remove loads of addresses within the allocation.
+  if (auto *Load = dyn_cast<LoadInst>(Inst))
```

Comment:
I'm wondering why we can't remove load instructions in general, i.e. remove the "!Inst->mayReadFromMemory()" from the condition above.